ledger/light/integration/light_test.go

```diff
@@ -57,6 +59,148 @@ func Test_BootstrapCalls(t *testing.T) {
 	})
 }
 
+func Test_LightReplication(t *testing.T) {
+	t.Parallel()
+
+	var secondPulseNumber = insolar.FirstPulseNumber + (PulseStep * 2)
+	var sentLifeline record.Lifeline
+	var sentObjectID insolar.ID
+
+	var sentIds = make(map[insolar.ID]struct{})
+	var replicationChannel = make(chan payload.Replication, 10)
+
+	ctx := inslogger.WithLoggerLevel(inslogger.TestContext(t), insolar.ErrorLevel)
+	cfg := DefaultLightConfig()
+
+	s, err := NewServer(ctx, cfg, func(meta payload.Meta, pl payload.Payload) {
+		switch pl.(type) {
+		case *payload.Replication:
+
+			if pl.(*payload.Replication).Pulse == secondPulseNumber {
+				go func() {
+					replicationChannel <- *pl.(*payload.Replication)
+				}()
+			}
+
+		}
+	})
+
+	require.NoError(t, err)
+
+	// First pulse goes in storage then interrupts.
+	s.Pulse(ctx)
+
+	// Second pulse goes in storage and starts processing, including pulse change in flow dispatcher.
+	s.Pulse(ctx)
+
+	t.Run("happy light replication", func(t *testing.T) {
+		// Creating root reason request.
+		var reasonID insolar.ID
+		{
+			p, _ := callSetIncomingRequest(ctx, t, s, gen.ID(), gen.ID(), record.CTSaveAsChild)
+			requireNotError(t, p)
+			reasonID = p.(*payload.RequestInfo).RequestID
+		}
+
+		// Save and check code.
+		{
+			p, sent := callSetCode(ctx, t, s)
+			requireNotError(t, p)
+			payloadId := p.(*payload.ID).ID
+
+			p = callGetCode(ctx, t, s, p.(*payload.ID).ID)
```

Comment:
Let's remove all from here to the end of block (except `sentIds[payloadId] = struct{}{}`).